specs/beacon-chain.md

```diff
@@ -324,36 +324,39 @@ def shuffle(values: List[Any],
     """
     values_count = len(values)
 
-    # entropy is consumed in 3 byte chunks
-    # sample_max is defined to remove the modulo bias from this entropy source
-    sample_max = 2 ** 24
-    assert values_count <= sample_max
+    # Entropy is consumed from the seed in 3-byte (24 bit) chunks.
+    rand_max = 2 ** 24 - 1
+    assert values_count < rand_max
 
     output = [x for x in values]
     source = seed
     index = 0
-    while index < values_count:
+    while index < values_count - 1:
         # Re-hash the source
         source = hash(source)
```

Comment:
Out of curiosity, why are we rehashing the source? Isn't it sufficient to use `seed` as is?

Reply:
The seed is re-hashed on two occasions:

1. On the first iteration of the shuffle.
2. Each time the seed is exhausted (i.e., we perform more than `floor(32 / 3)` iterations).

Assuming the seed initially supplied to the function is 32-bytes of entropy, I don't see an issue with removing the first (1) rehash.